docs/proposals/2018-11-01_-Alt_Credits.md

```diff
@@ -0,0 +1,89 @@
+# Problem
+
+On certain stories, newsroom users want to override or hide the photographer and credit names that are set in the downstream display channels they own (i.e., website, AMP, and native apps). 
+
+However, this information should not be modified in all downstream contexts. Shared stories over wires should contain the original credits for legal and other reasons. Consumers should be able to choose whether to use the alternate credits or not. 
+
+# Proposal
+
+Add an 'alternate_credits' trait on all ANS elements, also explicitly suggested usage within `referent_properties`.
+
+Consuming templates should be updated to use the `alternate_credits` trait if it is available and they desire to use it.
+
+
+## Alternate Credits Trait
+
+* `alternate_credits` will have the same structure as `credits` but potentially different data. 
+
+
+## Example Usage
+
+Before denormalization.
+
+```
+{
+  "type": "story",
+  "version": "0.8.0",
+   
+  "content_elements": [
+     {
+       "type": "reference",
+       "referent": {
+         "type": "image",
+         "id": "DEFGHIJKLMNOPQRSTUVWXYZABC",
+         "referent_properties": {
+           "alternate_credits": {
+             "by": [
+               {
+                 "type": "author",
+                 "version": "0.8.0",
+                 "byline": "If you do not wish to display the actual photographer, you can display this message instead."
+               }
+             ]
+           }
+         }
+       }
+     }
+  ]
+}
+```
+
+After denormalization.
+
+```
+{
+  "type": "story",
+  "version": "0.8.0",
+   
+  "content_elements": [
+    {
+      "type": "image",
+      "_id": "DEFGHIJKLMNOPQRSTUVWXYZABC",
+      "version": "0.8.0",
+      "url": "https://www.washingtonpost.com/foo.jpg",
+      "credits": {
+        "by": [
+          {
+            "type": "author",
+            "version": "0.8.0",
+            "byline": "Jim Phelps"
+          }
+        ]
+      },  
+      "alternate_credits": {
```

Comment:
How does this work if we have multiple alternate credits?

Reply:
Do you mean, if there are alternates for more than one context? (E.g., show this on iPhone, this on iPad, this on web, this on the feed...) This proposal does not support that.

Reply:
Right. Or another example would be www, rainbow, apple news, the lily. And this is a pretty real scenario, though most of those are not using ANS directly yet.

I think this is too fragile. If we want to support multiple bylines, we should be supporting for `n`.

Reply:
If you're talking about the rainbow use case, I think that needs a much, much broader solution -- potentially any field needs an alternate, over some (hopefully enumerated) set of contexts.  I'd rather solve that with generic overrides or story branching than individual "alternate" fields.  This is intended as a point solution for a very specific use case -- wanting to vary the "nice presentation" of certain embedded images from the "real" version that we send downstream to consumers.

One point that this does raise is that "alternate" is too vague a name -- we should probably be explicit about exactly where this set of credits should be used instead of the regular set.

@chadmoone Does this answer alleviate your concerns at all?

Reply:
> One point that this does raise is that "alternate" is too vague a name -- we should probably be explicit about exactly where this set of credits should be used instead of the regular set.

This is a good point. I don't know if the focus is more on "vanity" or "short," but I appreciate precision of specification.

Reply:
Updated proposal to better describe the feature, and change the name to `preferred_credits`. I'm not sure it's all *that* much better.

Other options:

- `native_credits`
- `local_credits`
- `render_credits`
- `reader_credits`
- `curated_credits`

I hate all of these.

Reply:
Alright, I guess (as mentioned in slack) I'd just prefer a more precise name if possible. Of course naming things is hard, so whatever we can get consensus around is fine.